alignment_check_experiment/01_umap_parameter_search.py

```diff
@@ -0,0 +1,243 @@
+    open(f"{str(decade_folder)}/aligned_word_vectors_2000_2020.pkl", "rb")
+)
+
+year_comparison_dict = {
+    "_".join(comparison_file.stem.split("_")[0:2]): (
+        pd.read_csv(str(comparison_file), sep="\t")
+    )
+    for comparison_file in (
+        list(Path(f"{str(decade_folder)}/year_distances_2005_2020").rglob("*tsv"))
+    )
+}
+list(year_comparison_dict.keys())[0:3]
+
+year_comparison_dict["2005_2020"].sort_values("global_dist", ascending=False)
+
+# # Train the UMAP Model
+
+origin_df = aligned_models["2000"]
+word_vectors = list(
+    map(
+        lambda x: x.query(f"token in {origin_df.token.tolist()}")
+        .sort_values("token")
+        .set_index("token")
+        .values,
+        aligned_models.values(),
+    )
+)
+
+word_models_stacked = np.vstack(word_vectors)
+file_name = "output/2000_2020_umap_model"
+
+# ## Neighbors vs Min Distance
+
+n_neighbors = [15, 25]
+min_dist = [0.0, 0.1, 0.25, 0.5, 0.8, 0.99]
+
+for neighbor, min_d in itertools.product(n_neighbors, min_dist):
+
+    if Path(f"output/param_sweep_results/{neighbor}_{min_d}_results.tsv").exists():
+        continue
+
+    model = ParametricUMAP(
+        verbose=False,
+        metric="cosine",
+        random_state=100,
+        low_memory=True,
+        n_neighbors=neighbor,
+        min_dist=min_d,
+    )
+    embedding = model.fit_transform(word_models_stacked)
+    embedding_model_df = pd.DataFrame(
+        {
+            "umap1": embedding[:, 0],
+            "umap2": embedding[:, 1],
+            "year": itertools.chain(
+                *list(
+                    map(
+                        lambda x: aligned_models[x]
+                        .query(f"token in {origin_df.token.tolist()}")
+                        .sort_values("token")
+                        .set_index("token")
+                        .assign(year=x)
+                        .year.tolist(),
+                        aligned_models.keys(),
+                    )
+                )
+            ),
+            "token": itertools.chain(
+                *list(
+                    map(
+                        lambda x: x.query(f"token in {origin_df.token.tolist()}")
+                        .sort_values("token")
+                        .set_index("token")
+                        .index.tolist(),
+                        aligned_models.values(),
+                    )
+                )
+            ),
+        }
+    )
+
+    embedding_model_df.to_csv(
+        f"output/param_sweep_results/{neighbor}_{min_d}_results.tsv",
+        sep="\t",
+        index=False,
+    )
+
+grid_results = dict()
+for file in Path("output/param_sweep_results").rglob("*tsv"):
+    grid_results[tuple(file.stem.split("_")[0:2])] = pd.read_csv(str(file), sep="\t")
+
+plot_df = pd.DataFrame(
+    [], columns=["umap1", "umap2", "year", "token", "neighbor", "min_d"]
+)
+for key in grid_results:
+    plot_df = plot_df.append(
+        grid_results[key]
+        .query("year in [2000, 2005, 2015, 2020]")
```

Comment:
Is there a reason you skip 2010 in these plots? Based on the plots in the second notebook I'd expect it to fall in with 2015 and 2020, but it was a bit weird to see a list with every 5 years except 2010

Reply:
Honestly 2010 slipped my mind. I'll re-add to keep consistent.